modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/operation/ComponentMessageProcessorBuilder.java

```diff
@@ -153,9 +157,36 @@ protected ResolverSet getArgumentsResolverSet() throws ConfigurationException {
   }
 
   protected ConfigurationProvider getConfigurationProvider() {
-    return parameters.values().stream()
-        .filter(v -> v instanceof ConfigurationProvider)
-        .map(v -> ((ConfigurationProvider) v)).findAny()
-        .orElse(configurationProvider);
+    ConfigurationProvider explicitConfigProvider = getConfigurationProvider(parameters.get("config-ref"));
```

Comment:
Still need to find a better way of finding out for which parameters do we need to do all the `ValueResolver` considerations.

Reply:
There is a more up to date [comment](https://github.com/mulesoft/mule/pull/11413#discussion_r863039104) which is related to this issue. I'm marking this one as resolved so we can track it in the other one.